### PR TITLE
Split dependency, fog and visibility notifier

### PR DESCRIPTION
**Pull request by BastiaanOlij**
This splits out the rest of our logic from storage and retires the main storage classes. 

There is logic that is still in `render_scene_render` that should move into the fog class as well but I will look into that in a separate PR.

### CODE REVIEW

File: servers/rendering/renderer_rd/storage_rd/mesh_storage.h [hunk 1]
```diff
@@ -35,7 +35,8 @@
 #include "core/templates/rid_owner.h"
 #include "core/templates/self_list.h"
 #include "servers/rendering/renderer_rd/shaders/skeleton.glsl.gen.h"
-#include "servers/rendering/renderer_storage.h"
+// #include "servers/rendering/renderer_storage.h"
```
Comment (clayjohn) [inline servers/rendering/renderer_rd/storage_rd/mesh_storage.h:38]: Can just delete I suppose

File: servers/rendering/renderer_rd/storage_rd/particles_storage.h [hunk 2]
```diff
@@ -37,8 +37,9 @@
 #include "servers/rendering/renderer_rd/shaders/particles.glsl.gen.h"
 #include "servers/rendering/renderer_rd/shaders/particles_copy.glsl.gen.h"
 #include "servers/rendering/renderer_rd/storage_rd/material_storage.h"
-#include "servers/rendering/renderer_storage.h"
+// #include "servers/rendering/renderer_storage.h"
```
Comment (clayjohn) [inline servers/rendering/renderer_rd/storage_rd/particles_storage.h:40]: Just delete

**Review by clayjohn — COMMENTED** [review]
I would move ``set_default_clear_color`` and ``get_default_clear_color`` into texture_storage then move everything else into a new class under under storage called ``RendererUtilities``

Everything else looks great!

File: servers/rendering/renderer_rd/storage_rd/mesh_storage.h (hunk 1 above)
Comment (BastiaanOlij) [reply servers/rendering/renderer_rd/storage_rd/mesh_storage.h:38]: owh yes I need to thanks :)

**Comment by BastiaanOlij** [comment]
> I would move `set_default_clear_color` and `get_default_clear_color` into texture_storage then move everything else into a new class under under storage called `RendererUtilities`
> 
> Everything else looks great!

Oh dang, i was planning to do that for default color. There are a few utility functions in some of the subclasses that I think can move into material storage as well

I like the idea of introducing `RendererUtilities`. I actually think that should include dependencies as well.

**Comment by BastiaanOlij** [comment]
@clayjohn you may want to have a closer look at the changes in GLES3, there was a bunch of helper and old commented out stuff in storage, I've moved those to where I think they make sense to park them but as there seemed to be some dead code I may have broken something.

File: drivers/gles3/storage/texture_storage.h [hunk 3]
```diff
@@ -364,6 +375,10 @@ class TextureStorage : public RendererTextureStorage {
 
 	RID_Owner<CanvasTexture, true> canvas_texture_owner;
 
+	/* CANVAS SHADOW */
+
+	RID_PtrOwner<CanvasLightShadow> canvas_light_shadow_owner;
```
Comment (clayjohn) [inline drivers/gles3/storage/texture_storage.h:380]: I'm not sure if texture_storage is the right place for shadow textures, like this. But I haven't implemented 2D lights yet, so its fine to place here and when I get to 2D lights I can move it if necessary.

**Review by clayjohn — COMMENTED** [review]
This is looking great so far!

File: drivers/gles3/storage/texture_storage.h (hunk 3 above)
Comment (BastiaanOlij) [reply drivers/gles3/storage/texture_storage.h:380]: I wasn't sure either, had to place it somewhere, open to suggestions or we can just leave it there for now seeing this so far seems to impact GLES3 only

Comment (clayjohn) [reply drivers/gles3/storage/texture_storage.h:380]: Yeah, I'm not even sure I will be using a CanvasLightShadow struct like that. Let's just leave it there for now.

**Review by clayjohn — APPROVED** [review]
Looks good to me!

**Review by reduz — APPROVED** [review]

**Comment by reduz** [comment]
Fantastic work Bastiaan!

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**